Various build tweaks

ddn0 opened the pull request.

ddn0 commented on lonestar/querying/cpu/querytest/CMakeLists.txt:
```diff
@@ -6,6 +6,20 @@ set(samples ${BASEINPUT}/cypherqueries/)
+list(APPEND threads "1")
+
+# add tests
+foreach (thr ${threads})
```
sorry, late to the game, but would the atomic increment issue also manifested itself for the smaller inputs that we have in the CI tests?

Also if you want to support parallel test execution, you'll have to play around with CMake test fixtures and dependencies (see lonestar/CMakeLists.txt and tools/CMakeLists.txt for examples).

l-hoang replied:
For the query test in particular? Nope, because I was never running multi-thread for CI in the first place.

ddn0 committed:
build: Avoid CMake COMMAND_ERROR_IS_FATAL

COMMAND_ERROR_IS_FATAL was introduced in CMake 3.19. The minimum CMake
version supported by the build is 3.17.

ddn0 committed:
build: Add pyarrow includes for cythonize

`conda install pyarrow` places header files under $PREFIX while `pip
install pyarrow` places header files under site-packages/pyarrow, which
is not a normal C++ include path and must be included explicitly.

arthurp commented on cmake/Modules/KatanaPythonSetupSubdirectory.cmake:
```diff
@@ -227,20 +227,24 @@ function(add_python_setuptools_target TARGET_NAME)
 
   # TODO(amp): The RPATH of the installed python modules will contain a reference to the build
-  #  directory. This is not ideal and could cause confusion, but without depending on an
-  #  additional command line tool there isn't any real way to fix it.
```
To explain: the initial space makes my IDE view all the lines as a single Todo. Not important, but something you will find on most of my todos.

arthurp commented:
A warning: pip pyarrow includes it's own libarrow so things will get linked funny. And might break things in a pretty serious ways of someone tries to use pyarrow along with katana arrow arrays.
